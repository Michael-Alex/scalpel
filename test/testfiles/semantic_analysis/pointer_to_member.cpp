class c
{
};

namespace n
{
	class c
	{
	};

	void (c::*fun_ptr)(int);
	int ::c::* int_ptr;
}

void f(int c::* int_ptr)
{
}

