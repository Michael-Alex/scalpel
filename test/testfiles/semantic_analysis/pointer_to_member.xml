<namespace id="n0">
 <namespace name="n" id="n1">
  <class name="c" id="c0">
   <destructor access="public" defined="false">
   </destructor>
  </class>
  <variable name="fun_ptr">
   <type>
    <pointer_to_member class_id="c0">
     <function_type>
      <return_type>
       <fundamental_type type="void"/>
      </return_type>
      <parameter_types>
       <fundamental_type type="int"/>
      </parameter_types>
     </function_type>
    </pointer_to_member>
   </type>
  </variable>
  <variable name="int_ptr">
   <type>
    <pointer_to_member class_id="c1">
     <fundamental_type type="int"/>
    </pointer_to_member>
   </type>
  </variable>
 </namespace>
 <class name="c" id="c1">
  <destructor access="public" defined="false">
  </destructor>
 </class>
 <simple_function name="f">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
  <parameters>
   <parameter name="int_ptr">
    <type>
     <pointer_to_member class_id="c1">
      <fundamental_type type="int"/>
     </pointer_to_member>
    </type>
   </parameter>
  </parameters>
 </simple_function>
</namespace>
