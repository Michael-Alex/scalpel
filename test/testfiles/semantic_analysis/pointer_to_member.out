<namespace id="0">
 <namespace id="1" name="n">
  <class id="0" name="c">
   <destructor access="public" defined="false">
   </destructor>
  </class>
  <variable id="0" name="fun_ptr">
   <type>
    <pointer_to_member  class_id="0">
     <function_type>
      <return_type>
       <fundamental_type type="void"/>
      </return_type>
      <parameter_types>
       <fundamental_type type="int"/>
      </parameter_types>
     </function_type>
    </pointer_to_member>
   </type>
  </variable>
  <variable id="1" name="int_ptr">
   <type>
    <pointer_to_member  class_id="1">
     <fundamental_type type="int"/>
    </pointer_to_member>
   </type>
  </variable>
 </namespace>
 <class id="1" name="c">
  <destructor access="public" defined="false">
  </destructor>
 </class>
 <simple_function id="0" name="f">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
  <parameters>
   <parameter name="int_ptr">
    <type>
     <pointer_to_member  class_id="1">
      <fundamental_type type="int"/>
     </pointer_to_member>
    </type>
   </parameter>
  </parameters>
 </simple_function>
</namespace>
